ChainSafe/lodestar
# feat: pre-electra support from attestation pool

- Attestation pool
    - Add type check to ensure adding and getting correct attestation type according to the fork
    - `aggregateByIndexByRootBySlot` is modified to allow null committee index to only handle pre-electra attestations. null committee index is not allowed post-electra attestations
- Aggregated attestation pool
    - Add type check to ensure adding and getting correct attestation type according to the fork
- Gossip
     - `beacon_attestation` in gossip queue is indexed by attestation data instead of attestation data + committee bits.
- API
     - Adjust implementations of several APIs to handle correct attestation type: getPoolAttestations, getPoolAttestationsV2, getAggregatedAttestation
